__tests__/app.test.js

```diff
@@ -4,40 +4,88 @@ const db = require('../db/connection');
 const testData = require('../db/data/test-data')
 const seed = require('../db/seeds/seed')
 
+// ---- connect and disconnect for each test ----
 beforeEach(() => {
-  return seed(testData)
+  return seed(testData) 
 })
 
 afterAll(() => {
-  return db.end();
-});
-// 404 error before or after?
-describe('GET /api/topics', () => {
-  test('status 200: returns an array of topic objects with a key of slug and description', () => {
-     return request(app)
-     .get("/api/topics")
-     .expect(200)
-     .then((res) => {
-      expect(res.body.topics.length).toBe(3);     //topics to have 3 topics
-      res.body.topics.forEach((topic) => {
-        expect(topic).toEqual(expect.objectContaining({
-          slug: expect.any(String),
-          description: expect.any(String),
-        }))  // minimum contains these prop
-      } )       // contain slug and description
-     })
-  })
+  return db.end(); 
 });
 
-//tested loosely- when using id etc need to be more specific
 
-describe('errors', () => {
-  test('status: 404 - returns a path not found message', () => {
+describe('TOPICS', () => {
+  // ---- GETS ----
+  describe('GET /api/topics', () => {
+    test('status 200: returns an array of topic objects with a key of slug and description', () => {
+       return request(app)
+       .get("/api/topics")
+       .expect(200)
+       .then((res) => {
+        expect(res.body.topics.length).toBe(3); 
+        res.body.topics.forEach((topic) => {
+          expect(topic).toEqual(expect.objectContaining({
+            slug: expect.any(String),
+            description: expect.any(String),
+          })) 
+        } )   
+       })
+    })
+  });
+// ---- ERRORS ----
+describe('ERRORS /api/topics', () => {
+  test(`status: 404 - returns a path not found message if topic doesn't exist`, () => {
     return request(app)
       .get('/api/jibberish')
       .expect(404)
-      .then((response) => {
-        expect(response.body.message).toBe('path not found');
+      .then((res) => {
+        expect(res.body.message).toBe('Path not found.');
+      });
+  });
+});
+});
+
+
+describe('ARTICLE_ID', () => {
+  // ---- GETS ----
+  describe('GET /api/articles/:article_id', () => {
+    test('status 200: returns an article object with specified properties', () => {
+      return request(app)
+      .get('/api/articles/1')
+      .expect(200)
+      .then(({body: {article}}) => {
+          expect(article).toBeInstanceOf(Object)
+          expect(article).toEqual(
+              expect.objectContaining({
+                  author: expect.any(String),
+                  title: expect.any(String),
+                  article_id: expect.any(Number),
+                  body: expect.any(String),
+                  topic: expect.any(String),
+                  created_at: expect.any(String),
+                  votes: expect.any(Number)
```

Comment:
As a specific article is being requested here - we should be able to expect a specific article on the response body - we should therefore be asserting for specific values on this object. Still keep `expect.objectContaining` in place to account for additional properties being added in the future - but we should know the author etc.
- You can keep `created_at: expect.any(String)` as is 👍

Reply:
aaah! point missed on that one! Updated :) Thanks